### PR TITLE
MM-17488: simplify plugin health check

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Simplify the plugin health check to only leverage the `Ping` API instead of sending signals, the latter of which is not supported on Windows.

Verified this continues working with a plugin that `os.Exit(1)`s ten seconds after startup.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17488

### CODE REVIEW

File: plugin/supervisor.go [hunk 1]
```diff
@@ -111,13 +108,8 @@ func (sup *supervisor) Hooks() Hooks {
 	return sup.hooks
 }
 
-// PerformHealthCheck checks the plugin through a process check, an RPC ping, and a HealthCheck hook call.
+// PerformHealthCheck checks the plugin through an an RPC ping.
```
Comment (lieut-data) [inline plugin/supervisor.go]: Did we ever have a `HealthCheck` call?

Comment (mickmister) [reply plugin/supervisor.go]: @lieut-data Yes, it was originally part of the PR that introduced the health check, but there was not a consensus on how it should work, so it was removed. See https://github.com/mattermost/mattermost-server/pull/10462#discussion_r279267206. My apologies for leaving this comment this way.

Comment (lieut-data) [reply plugin/supervisor.go]: No worries! Just wanted to make sure I didn't miss anything :D

**Comment by lieut-data** [comment]
@prapti, adding you as an assignee just to confirm this is likely an issue we want to verify post-merge.

**Comment by prapti** [comment]
@lieut-data Looks like it needs to be verified on Windows env which I don't have. I can ask around in QA to see if anyone would like this post merge?

**Comment by lieut-data** [comment]
That would be great, @prapti! Let us know when we can merge :)

**Comment by lindalumitchell** [comment]
This one can be tested after it's merged rather than on the PR. Removing QA Review label.

**Pull request merged**